### DISCUSSION TITLE
Contact Group linking to Objects

### DISCUSSION THREAD

**Discussion by ryanmerolle** [netbox-community/netbox#8359] [General]
Currently, a user has to add each contact for a company to each object (device, site, provider, manufacturer).  I think would it be remove some of the tediousness of this process to allow users to link contact groups to those objects.  Especially if people organize contact groups by companies or departments.

How do people currently handle this?  Would such a feature request be popular?

If such a feature was not introduced, would people just use #7006 to link contact groups to objects?

**Comment by ozmusic** [comment]
It makes sense to be able to add a group as a contact to an object instead of the individual team members

webservers have contact group SRE-WEB assigned
DB servers have the contact group SRE-DB assigned

Tom leaves the web team to join the DB team, one group membership change is needed to complete the move in netbox, instead of updating the individual objects

**Comment by ryanmerolle** [comment]
The challenge is that contact role is assigned after associating a contact to an object.  You would miss out on the contact role function given the current relationships.

**Comment by ITJamie** [comment]
I also want to be able to add an individual contact to multiple groups. In some cases this is needed where the IT contact is an contractor for multiple seperate companies. I would want to have a group for that company and add the it contact to it aswell.

**Reply by ryanmerolle** [reply]
Ah that makes things way more complex.  At first I was thinking this was very niche / rare, but the model of 1 contact to many contact groups could come up quite a bit beyond just contractors.

Too bad we did not suggest these relationships when the feature was requested in #1344

I am doubtful its worth changing much with this feature, but its a good discussion.

**Comment by hagbarddenstore** [comment]
Manage the group outside of Netbox.

In my department, we’re split up into multiple groups managing different parts of the platform, Linux OS, Windows OS, Hardware, Networking, etc. These groups exists in Office365 and have their own email address. We simply add that email as a contact in Netbox. Voila, the management of the group itself is outsourced.

**Reply by ryanmerolle** [reply]
Good approach, but it does not help as much for outside vendors when you are using NetBox to track support contacts.

**Comment by ozmusic** [comment]
Is it possible to have the contact selector list both contacts and contact groups?

The way it is set up at the moment, we would just do as @hagbarddenstore mentioned, for internal contacts a single contact would map to a distribution list, with a custom field for slack/teams channel, and another for opsgenie/pagerduty. For external contacts, it'd still be good to assign as a group, the alternative being to edit the existing contact to replace all the contact info when the supplier contact is changed so that the existing assignments are maintained.

**Reply by ryanmerolle** [reply]
That’s the sort of input and feedback this discussion is gathering before raising a feature request.

**Comment by VictorPavlushin** [comment]
I also would like to be able to assign a group as a contact. 
Then groups could be synchronized from external sources such as Active Directory.  
For example, a team has an access to certain servers of a certain information system and having a common group in AD, so this group can be synchronized into contacts.

**Comment by ITJamie** [comment]
being able to link a contact / contact group to a tenant might solve a lot of issues in future too. 
for example. some org needs another contact added to all objects. well if devices/racks etc all have the tenant assigned it just needs to be updated in one place

**Comment by empusas** [comment]
So from my point of view the proper way would be to have contacts/person that can be part of a group.
The group should have a role assigned, not the relation between a contact/group and an object.
Then contacts/person can be assigned to one or many groups. 
An object like tenant, device etc. can have one or multiple groups assigned. No need to have the role here again as it is already an attribute of the group.

**Comment by DCAuto** [comment]
Pinging this group to see if there has been a feature request submitted and where I can track that?

**Comment by mcdonalm-at-frib** [comment]
This discussion is older (though received more responses) than this other one I found talking about a few similar items:

https://github.com/netbox-community/netbox/discussions/15064

Would love to see some traction on this idea as I feel it could be very useful.

**Reply by michaelmcdonald** [reply]
It's been a year, just tossing another comment on this that the concept is super useful.
